jobs/wf2-2-upsertTEIs.js

```diff
@@ -12,69 +12,77 @@ fn(state => {
+    patient.identifiers.find(i => i.identifierType.uuid === uuid);
 
-  const [openMRSID] = patient.identifiers.filter(
-    i => i.identifierType.uuid === '05a29f94-c0ed-11e2-94be-8c13b969e334'
-  ); //finding OpenMRS ID that was auto-assigned
-  const identifier = OldId ? OldId.identifier : openMRSID.identifier;
-  
-  const dateCreated = patient.auditInfo.dateCreated.substring(0,10);
+  const patientsMapping = state.patients.map(patient => {
+    const oldId = pluckIdentifier(
+      patient,
+      '8d79403a-c2cc-11de-8d13-0010c6dffd0f'
+    ); //finding DHIS2 patient_number
+    const openMRSID = pluckIdentifier(
+      patient,
+      '05a29f94-c0ed-11e2-94be-8c13b969e334'
+    ); //finding OpenMRS ID that was auto-assigned
 
-  //   console.log(`Identifier: ${identifier}`);
-  //   console.log(`UUID: ${patient.uuid}`);
-  //   console.log(`Age: ${patient.person.age}`);
-  //   console.log(`Gender: ${genderOptions[patient.person.gender]}`);
+    const identifier = oldId ? oldId.identifier : openMRSID.identifier;
 
-  return upsert(
-    'trackedEntityInstances',
-    {
-      ou: 'l22DQq4iV3G',
-      filter: [`jGNhqEeXy2L:Eq:${patient.uuid}`],
-    },
-    {
-      program: 'uGHvY5HFoLG',
-      orgUnit: 'l22DQq4iV3G',
-      trackedEntityType: 'cHlzCA2MuEF',
-      attributes: [
-        {
-          attribute: 'P4wdYGkldeG',
-          value: identifier,
-        },
-        {
-          attribute: 'jGNhqEeXy2L',
-          value: patient.uuid,
-        },
-        {
-          attribute: 'qptKDiv9uPl',
-          value: genderOptions[patient.person.gender],
-        },
-        {
-          attribute: 'T1iX2NuPyqS',
-          value: patient.person.age,
-        },
-      ],
-      enrollments: [
-        {
+    const dateCreated = patient.auditInfo.dateCreated.substring(0, 10);
+
+    return {
+      query: {
+        ou: 'l22DQq4iV3G',
+        filter: [`jGNhqEeXy2L:Eq:${patient.uuid}`],
+      },
+      data: {
+        program: 'uGHvY5HFoLG',
+        orgUnit: 'l22DQq4iV3G',
+        trackedEntityType: 'cHlzCA2MuEF',
+        attributes: [
+          {
+            attribute: 'P4wdYGkldeG',
+            value: identifier,
+          },
+          {
+            attribute: 'jGNhqEeXy2L',
+            value: patient.uuid,
+          },
+          {
+            attribute: 'qptKDiv9uPl',
+            value: genderOptions[patient.person.gender],
+          },
+          {
+            attribute: 'T1iX2NuPyqS',
+            value: patient.person.age,
+          },
+        ],
+        enrollments: [
+          {
             orgUnit: 'l22DQq4iV3G',
             program: 'uGHvY5HFoLG',
             programStage: 'hfKSeo6nZK0',
-            enrollmentDate: dateCreated
-        }
-      ]
-    }
-  )(state);
+            enrollmentDate: dateCreated,
+          },
+        ],
+      },
+    };
+  });
+  return { ...state, patientsMapping };
 });
 
+// Upsert TEIs to DHIS2
+each(
+  'patients[*]',
```

Comment:
Shouldn't this be `each('patientsMapping[*]'...)` ?

Reply:
And rather than call it `patientsMapping`, which is kinda meaningless, maybe call it `patientsUpsert` or `teiUpsert` or something a bit richer